turbinia/workers/__init__.py

```diff
@@ -571,14 +574,26 @@ def execute(
 
     # Execute the job on the host system.
     else:
-      if shell:
-        proc = subprocess.Popen(
-            cmd, shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
-        proc.wait(timeout_limit)
-      else:
-        proc = subprocess.Popen(
-            cmd, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
-        proc.wait(timeout_limit)
+      try:
+        if shell:
+          proc = subprocess.Popen(
+              cmd, shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+          proc.wait(timeout_limit)
+        else:
+          proc = subprocess.Popen(
+              cmd, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+          proc.wait(timeout_limit)
+      except subprocess.TimeoutExpired as exception:
+        # Log error and close result.
+        message = 'Execution of [{0!s}] failed due to job timeout of seconds has been reached.'.format(
```

Comment:
Did you mean to add the actual seconds value here?

Reply:
No because the timeout_limit would be the actual seconds since it shouldn't run beyond this.